File: src/client/java/fr/hardel/asset_editor/client/javafx/routes/StudioRoute.java
package fr.hardel.asset_editor.client.javafx.routes;

public enum StudioRoute {
    ENCHANTMENT_OVERVIEW("enchantment", true, false),
    ENCHANTMENT_MAIN("enchantment", false, true),
    ENCHANTMENT_FIND("enchantment", false, true),
    ENCHANTMENT_SLOTS("enchantment", false, true),
    ENCHANTMENT_ITEMS("enchantment", false, true),
    ENCHANTMENT_EXCLUSIVE("enchantment", false, true),
    ENCHANTMENT_TECHNICAL("enchantment", false, true),
    ENCHANTMENT_SIMULATION("enchantment", false, false),
    ENCHANTMENT_DNT("enchantment", false, false),
    ENCHANTMENT_YGGDRASIL("enchantment", false, false),
    LOOT_TABLE_OVERVIEW("loot_table", true, false),
    LOOT_TABLE_MAIN("loot_table", false, true),
    LOOT_TABLE_POOLS("loot_table", false, true),
    RECIPE_OVERVIEW("recipe", true, false),
    RECIPE_MAIN("recipe", false, true),
    CHANGES_MAIN("changes", false, false);

    private final String concept;
    private final boolean overview;
    private final boolean tabRoute;

    StudioRoute(String concept, boolean overview, boolean tabRoute) {
        this.concept = concept;
        this.overview = overview;
        this.tabRoute = tabRoute;
    }

    public String concept() {
        return concept;
    }

    public boolean isOverview() {
        return overview;
    }

    public boolean isTabRoute() {
        return tabRoute;
    }

    public static StudioRoute overviewOf(String concept) {
        return switch (concept) {
            case "loot_table" -> LOOT_TABLE_OVERVIEW;
            case "recipe" -> RECIPE_OVERVIEW;
            default -> ENCHANTMENT_OVERVIEW;
        };
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/routes/StudioRoute.kt
package fr.hardel.asset_editor.client.compose.routes

enum class StudioRoute(
    private val conceptKey: String,
    private val overview: Boolean,
    private val tabRoute: Boolean
) {
    ENCHANTMENT_OVERVIEW("enchantment", overview = true, tabRoute = false),
    ENCHANTMENT_MAIN("enchantment", overview = false, tabRoute = true),
    ENCHANTMENT_FIND("enchantment", overview = false, tabRoute = true),
    ENCHANTMENT_SLOTS("enchantment", overview = false, tabRoute = true),
    ENCHANTMENT_ITEMS("enchantment", overview = false, tabRoute = true),
    ENCHANTMENT_EXCLUSIVE("enchantment", overview = false, tabRoute = true),
    ENCHANTMENT_TECHNICAL("enchantment", overview = false, tabRoute = true),
    ENCHANTMENT_SIMULATION("enchantment", overview = false, tabRoute = false),
    LOOT_TABLE_OVERVIEW("loot_table", overview = true, tabRoute = false),
    LOOT_TABLE_MAIN("loot_table", overview = false, tabRoute = true),
    LOOT_TABLE_POOLS("loot_table", overview = false, tabRoute = true),
    RECIPE_OVERVIEW("recipe", overview = true, tabRoute = false),
    RECIPE_MAIN("recipe", overview = false, tabRoute = true),
    CHANGES_MAIN("changes", overview = false, tabRoute = false),
    DEBUG("debug", overview = false, tabRoute = false),
    NO_PERMISSION("none", overview = false, tabRoute = false);

    fun concept(): String = conceptKey

    fun isOverview(): Boolean = overview

    fun isTabRoute(): Boolean = tabRoute

    companion object {
        @JvmField
        val EnchantmentOverview = ENCHANTMENT_OVERVIEW

        @JvmField
        val EnchantmentMain = ENCHANTMENT_MAIN

        @JvmField
        val EnchantmentFind = ENCHANTMENT_FIND

        @JvmField
        val EnchantmentSlots = ENCHANTMENT_SLOTS

        @JvmField
        val EnchantmentItems = ENCHANTMENT_ITEMS

        @JvmField
        val EnchantmentExclusive = ENCHANTMENT_EXCLUSIVE

        @JvmField
        val EnchantmentTechnical = ENCHANTMENT_TECHNICAL

        @JvmField
        val EnchantmentSimulation = ENCHANTMENT_SIMULATION

        @JvmField
        val LootTableOverview = LOOT_TABLE_OVERVIEW

        @JvmField
        val LootTableMain = LOOT_TABLE_MAIN

        @JvmField
        val LootTablePools = LOOT_TABLE_POOLS

        @JvmField
        val RecipeOverview = RECIPE_OVERVIEW

        @JvmField
        val RecipeMain = RECIPE_MAIN

        @JvmField
        val ChangesMain = CHANGES_MAIN

        @JvmField
        val Debug = DEBUG

        @JvmField
        val NoPermission = NO_PERMISSION

        @JvmStatic
        fun overviewOf(concept: String): StudioRoute = when (concept) {
            "loot_table" -> LOOT_TABLE_OVERVIEW
            "recipe" -> RECIPE_OVERVIEW
            else -> ENCHANTMENT_OVERVIEW
        }
    }
}
